packages/blade/src/components/Radio/_decisions/decisions.md

```diff
@@ -0,0 +1,113 @@
+| labelPosition      | `top \| left`                               | `top`       | The position of the rendered label                                                                                               |          |
+| defaultValue       | `string`                                    | `undefined` | The initial value of the radio group                                                                                             |          |
+| value              | `string`                                    | `undefined` | The value of the radio, if present will act as [controlled component](https://reactjs.org/docs/forms.html#controlled-components) |          |
+| onChange           | `({ value: string, name: string }) => void` | `undefined` | The function to be called when any radio's state changes                                                                         |          |
+| name               | `string`                                    | `undefined` | The name of the radio group, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name)  |          |
+| isDisabled         | `boolean`                                   | `false`     | Control whether the radio group is disabled or not.                                                                              |          |
+| helpText           | `string`                                    | `undefined` | The helper text to be rendered                                                                                                   |          |
+| errorText          | `string`                                    | `undefined` | The error text to be rendered                                                                                                    |          |
+| validationState    | `'none' \| 'error'`                         | `none`      | Control whether the radio group is invalid or not.                                                                               |          |
+| necessityIndicator | `'optional' \| 'required' \| 'none'`        | `none`      | Renders `${label} (optional)` for `optional` and `${label} *` for `required`. if set to `none` renders nothing                   |          |
+
+### Examples:
+
+#### Basic
+
+```tsx
+<RadioGroup
+  name="developers"
+  label="Developers"
+  helpText="Pick preferred blade developer"
+  defaultValue="anurag"
+>
+  <Radio value="anurag">Anurag</Radio>
+  <Radio value="chaitanya">Chaitanya</Radio>
+  <Radio value="divyanshu">Divyanshu</Radio>
+  <Radio value="kamlesh">Kamlesh</Radio>
+</RadioGroup>
+```
+
+#### Controlled, Uncontrolled
+
+```tsx
+const Controlled = () => {
+  const [selected, setSelected] = React.useState('kamlesh');
+
+  return (
+    <RadioGroup
+      name="developers"
+      label="Developers (controlled)"
+      value={selected}
+      onChange={setSelected}
```

Comment:
This wouldn't work right? Checking the params received by `onChange` callback 🤔

Reply:
Will fix.